ext/vl53/vl53_transport.hpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2025, Henrik Hose
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_VL53_TRANSPORT_HPP
+#define MODM_VL53_TRANSPORT_HPP
+
+#include <array>
+#include <cstdint>
+#include <modm/architecture/interface/delay.hpp>
+#include <modm/architecture/interface/i2c_device.hpp>
+#include <modm/architecture/interface/spi_device.hpp>
+#include <modm/processing/fiber.hpp>
+#include <optional>
+#include <span>
+
+namespace modm
+{
+
+class Vl53TransportBase
+{
+public:
+	virtual ~Vl53TransportBase() = default;
+
+	virtual uint8_t
+	writeMulti(uint16_t reg, uint8_t *values, uint32_t size) = 0;
+	virtual uint8_t
+	readMulti(uint16_t reg, uint8_t *values, uint32_t size) = 0;
+	virtual uint8_t
+	writeByte(uint16_t reg, uint8_t value) = 0;
+	virtual uint8_t
+	readByte(uint16_t reg, uint8_t *value) = 0;
+	virtual uint8_t
+	resetSensor() = 0;
+};
+
+template<typename SpiMaster, typename Cs, typename Lpn, size_t VL53LMZ_COMMS_CHUNK_SIZE = 4096>
+class Vl53SpiTransport : public modm::SpiDevice<SpiMaster>, public Vl53TransportBase
+{
+public:
+	Vl53SpiTransport()
+	{
+		this->attachConfigurationHandler([] {
+			SpiMaster::setDataMode(SpiMaster::DataMode::Mode3);
+			SpiMaster::setDataOrder(SpiMaster::DataOrder::MsbFirst);
+			SpiMaster::setDataSize(SpiMaster::DataSize::Bit8);
+		});
+	};
+
+	Vl53SpiTransport(const Vl53SpiTransport &) = delete;
+
+	Vl53SpiTransport &
+	operator=(const Vl53SpiTransport &) = delete;
+
+	uint8_t
+	writeMulti(uint16_t register_address, uint8_t *p_values, uint32_t size) override;
+
+	uint8_t
+	readMulti(uint16_t register_address, uint8_t *p_values, uint32_t size) override;
+
+	inline uint8_t
```

Comment:
```suggestion
	uint8_t
```
No explicit `inline` required here. It's implicitly `inline` because it's a definition inside the class declaration body.

Reply:
fixed